Mathlib/Combinatorics/SimpleGraph/Turan.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Combinatorics.SimpleGraph.Clique
+
+/-!
+# The Turán graph
+
+This file defines the Turán graph and proves some of its basic properties.
+
+## Main declarations
+
+* `G.IsTuranMaximal r`: predicate saying that `G` has the most number of edges for its number `n`
+  of vertices while still being `r + 1`-cliquefree.
+* `turanGraph n r`: the canonical `r + 1`-cliquefree Turán graph on `n` vertices.
+
+## TODO
+
+* Port the rest of Turán's theorem from https://github.com/leanprover-community/mathlib4/pull/9317
+-/
+
+
+open Finset
+
+namespace SimpleGraph
+
+variable {V : Type*} [Fintype V] [DecidableEq V]
+variable (G : SimpleGraph V) [DecidableRel G.Adj]
+
+/-- An `r + 1`-cliquefree graph is `r`-Turán-maximal if any other `r + 1`-cliquefree graph on
+the same vertex set has the same or fewer number of edges. -/
+def IsTuranMaximal (r : ℕ) : Prop :=
+  G.CliqueFree (r + 1) ∧ ∀ (H : SimpleGraph V) [DecidableRel H.Adj],
+    H.CliqueFree (r + 1) → H.edgeFinset.card ≤ G.edgeFinset.card
+
+/-- The canonical `r + 1`-cliquefree Turán graph on `n` vertices. -/
+def turanGraph (n r : ℕ) : SimpleGraph (Fin n) where Adj v w := v % r ≠ w % r
+
+/-- `turanGraph n 0` is complete. Since the only 1-cliquefree graph is the empty graph,
+this shows that `0 < r` is needed in the statement of Turán's theorem. -/
+theorem turanGraph_zero_eq_top {n : ℕ} : turanGraph n 0 = ⊤ := by
+  ext a b
+  simp_rw [turanGraph, top_adj, Nat.mod_zero, not_iff_not, Fin.val_inj]
+
+variable {n r : ℕ}
+
+instance : DecidableRel (turanGraph n r).Adj := by dsimp only [turanGraph]; infer_instance
+
+/-- For `n ≤ r`, `turanGraph n r` is complete. -/
+theorem turanGraph_eq_top_of_le {n r} (h : n ≤ r) : turanGraph n r = ⊤ := by
```

Comment:
You're missing a theorem of the form
```
@[simp] theorem turanGraph_eq_top {n r} : turanGraph n r = ⊤ ↔ r = 0 ∨ n ≤ r := by
```
You might as well use it to replace `turanGraph_eq_top_of_le`

Reply:
Done.